Fix/guide layout

goodhoko opened the pull request:
@tomasklim Kindly asking for review. Read the commit message for details.

goodhoko commented on packages/suite/src/components/wallet/AccountsMenu/index.tsx:
```diff
@@ -134,7 +135,11 @@ const AccountsMenu = ({ device, accounts, selectedAccount }: Props) => {
     const { params } = selectedAccount;
-    const { isMobileLayout } = useLayoutSize();
+    const { isMobileLayout, layoutSize } = useLayoutSize();
+    // Must be kept in sync with SuiteLayout.
+    const isInline = useSelector(
+        state => isMobileLayout || (layoutSize !== 'XLARGE' && state.guide.open),
+    );
```
I don't like this. But I'm not sure how to do this better.

It would be IMO best to pass the `isInline` as a prop. However I didn't find out how that can be done in `SuiteLayout` where TS doesn't know that `menu` is actually `AccountsMenu`. I'm open to any suggestion on how to do this nicer.

tomasklim replied:
I think AccountsMenu should have an access to `LayoutContext.Provider`? So you can define `isInline` in the `SuiteLayout` file and pass it here. 

(I am not 100% sure, it is hard to get into this code without spending a lot of time on it.)

goodhoko replied:
@tomasklim Thanks a lot! I had to educate myself on what React Context is but now I think I cracked it. Please check it out.
03251f41d6cef02e44dd1a22ac4a53ac34ae85e9

goodhoko commented on packages/suite/src/components/suite/SuiteLayout/index.tsx:
```diff
@@ -141,6 +142,27 @@ const BodyWide = ({ url, menu, appMenu, children, guideOpen }: BodyProps) => (
     </Body>
 );
 
+/**
+ * Displays menu and body in two columns when guide is closed.
+ * Displays menu inlined in body when guide is open.
+ */
+const BodyNormal = ({ url, menu, appMenu, children, guideOpen }: BodyProps) => (
+    <Body>
+        <Columns>
+            {!guideOpen && menu && <MenuSecondary>{menu}</MenuSecondary>}
+            <ScrollAppWrapper url={url}>
+                {guideOpen && menu}
+                {appMenu}
+                <DefaultPaddings>
+                    <MaxWidthWrapper>{children}</MaxWidthWrapper>
+                </DefaultPaddings>
+            </ScrollAppWrapper>
+            {guideOpen && <StyledGuidePanel />}
+        </Columns>
+    </Body>
+);
+
```
Here (and below at lines 217-226) I traded a small code duplicity for more straight forward code and easier to read doc-comments. However, I'm open to any advice on how to make this more concise.

tomasklim replied:
I don't like the duplicity. However, it is true, that thanks to this, the magic here is more understandable.

I have an idea to merge `BodyWide` and `BodyNormal` into one function. Something like this (I am not 100% happy with it, needs some polishing):

```
// also you can exchange layoutSize for isXLargeLayout directly
const BodyDesktop = ({ url, menu, appMenu, children, guideOpen, layoutSize }: BodyProps) => {
    // your previous comments
    const isXLargeLayout = layoutSize === 'XLARGE';

    return (
        <Body>
            <Columns>
                {menu && (isXLargeLayout || (!isXLargeLayout && !guideOpen)) && (
                    <MenuSecondary>{menu}</MenuSecondary>
                )}
                <ScrollAppWrapper url={url}>
                    {!isXLargeLayout && guideOpen && menu}
                    {appMenu}
                    <DefaultPaddings>
                        <MaxWidthWrapper>{children}</MaxWidthWrapper>
                    </DefaultPaddings>
                </ScrollAppWrapper>
                {guideOpen && <StyledGuidePanel />}
            </Columns>
        </Body>
    );
};
```

tomasklim replied:
EDIT:
I do not want to delete the previous comment. But I think that you can merge it to 1 function and somehow use the `isInline` variable mentioned in the last comment.

goodhoko replied:
Kinda did here b63cf3eaad38f7c22987ad9c123a35f0607c15f1

If you think that's better, please resolve this. .)

tomasklim replied:
I did not test it, but it looks definitely better 👍

goodhoko reviewed (commented):
@tomasklim Below I highlighted parts I'm not satisfied with.

One more thing: I didn't write tests for this (yet). I'm wondering: how do we even test this? I guess I should somehow initiate cypress with a particular viewport resolution and then check for (dis)appearance of certain elements...? Or can we use the screenshot diffing caught sight of somewhere on Slack?

tomasklim commented on packages/suite/src/components/suite/SuiteLayout/index.tsx:
```diff
@@ -141,6 +142,27 @@ const BodyWide = ({ url, menu, appMenu, children, guideOpen }: BodyProps) => (
+);
+
+/** Displays menu inlined in body and hides guide. */
 const BodyNarrow = ({ url, menu, appMenu, children }: BodyProps) => (
```
```suggestion
const BodyMobile = ({ url, menu, appMenu, children }: BodyProps) => (
```

tomasklim replied:
Lets make it self-descriptive

goodhoko replied:
https://github.com/trezor/trezor-suite/pull/3887/commits/b64f41d4dea3b528d0a0f9c55122b624b9e14c14

I'll resolve this with your permission. .)

tomasklim commented on packages/suite/src/components/suite/SuiteLayout/index.tsx:
```diff
@@ -192,6 +214,16 @@ const SuiteLayout = (props: SuiteLayoutProps) => {
+                    </BodyNormal>
+                )}
                 {isMobileLayout && (
                     <BodyNarrow menu={menu} appMenu={appMenu} url={props.router.url}>
```
```suggestion
                    <BodyMobile menu={menu} appMenu={appMenu} url={props.router.url}>
```

goodhoko replied:
dtto https://github.com/trezor/trezor-suite/pull/3887/files#r649801110

tomasklim reviewed (commented):
It is working for me. Probably it can be merged but we should create an issue to refactor all these components because it is not very understandable. (at least for me)

tomasklim reviewed (approved):
Super. Great job! From my side, ready to be squashed and merged

matejkriz commented:
there is one small issue on screen width 1200 - ~1400 px

![responsiveness_issue](https://user-images.githubusercontent.com/3729633/121878883-820b1c80-cd0c-11eb-8628-636685136dfd.png)


I believe, that removing `white-space: nowrap;` from `SecondaryValueWrapper` in InfoCard would be good-enough fix for that...

goodhoko commented:
> there is one small issue on screen width 1200 - ~1400 px [...]

Thanks! [Fixed](https://github.com/trezor/trezor-suite/compare/a27af0b91dbad3448e267fe23410a06cd8e44dca..38700fba1da97639c3bc705fccda20925a45a4f5) by the suggested change.

I also rebased and squashed. So once the CI finishes I'll merge this.

tomasklim commented on packages/suite/src/views/wallet/transactions/components/TransactionSummary/components/SummaryCards/components/InfoCard/index.tsx:
```diff
@@ -47,7 +47,6 @@ const SecondaryValueWrapper = styled.div`
     font-variant-numeric: tabular-nums;
     /* margin-left: 1ch; */
     line-height: 1.57;
-    white-space: nowrap;
```
I would avoid separating date on more lines.

Instead of 
Feb 10, 2021 - Apr 17, 
2021

I would do
Feb 10, 2021 -
Apr 17, 2021

If you agree...

matejkriz replied:
great idea, but I'm afraid that it's surprisingly hard to achive with react-intl :-(

tomasklim replied:
This range is created from two `FormattedDate` components, so I think it should not be a problem

goodhoko replied:
Fixed in https://github.com/trezor/trezor-suite/pull/3887/commits/6cfc57127c9e4c265ae91ea484333ac72c439c76
